packages/snaps-controllers/src/snaps/SnapController.test.ts

```diff
@@ -925,25 +925,43 @@ describe('SnapController', () => {
 
   it('does not timeout while waiting for response from MetaMask', async () => {
     const sourceCode = `
-    module.exports.onRpcRequest = () => wallet.request({ method: 'eth_blockNumber', params: [] });
+    module.exports.onRpcRequest = () => ethereum.request({ method: 'eth_blockNumber', params: [] });
     `;
 
-    const [snapController, service] = getSnapControllerWithEES(
-      getSnapControllerWithEESOptions({
-        idleTimeCheckInterval: 30000,
-        maxIdleTime: 160000,
-        state: {
-          snaps: getPersistedSnapsState(
-            getPersistedSnapObject({
-              sourceCode,
-              manifest: getSnapManifest({
-                shasum: getSnapSourceShasum(sourceCode),
-              }),
+    const options = getSnapControllerWithEESOptions({
+      environmentEndowmentPermissions: [SnapEndowments.EIP1193],
+      idleTimeCheckInterval: 30000,
+      maxIdleTime: 160000,
+      state: {
+        snaps: getPersistedSnapsState(
+          getPersistedSnapObject({
+            sourceCode,
+            manifest: getSnapManifest({
+              shasum: getSnapSourceShasum(sourceCode),
             }),
-          ),
-        },
-      }),
-    );
+          }),
+        ),
+      },
+    });
+
+    const { rootMessenger } = options;
+
+    const originalCall = rootMessenger.call.bind(rootMessenger);
+
+    jest.spyOn(rootMessenger, 'call').mockImplementation((method, ...args) => {
+      // Give snap EIP1193 permission
+      if (method === 'PermissionController:hasPermission') {
+        return true;
+      } else if (
+        method === 'PermissionController:getEndowments' &&
+        args[1] === SnapEndowments.EIP1193
+      ) {
+        return ['ethereum'];
+      }
+      return originalCall(method, ...args) as any;
+    });
```

Comment:
It would be nice if we had a better way to add stuff like this to tests. For #905, I'm having to duplicate it a lot across different tests that mock the messenger.

Out of scope for this PR though.

Reply:
Yeah, I'm not sure what a good abstraction/pattern is for this though